### ISSUE TITLE
Implementation of in-app EXIF modification

### ISSUE THREAD

**Issue by ilgazer** [commons-app/apps-android-commons#1686]
**Summary:** 

The app should implement a group of global setting for EXIF modification.

**Expected behaviour:** 

In the settings menu, there should be a setting to remove certain EXIF tags like camera model etc(#70, expansion of #1685 ) This is partially implemented in #1685.

There should be another setting to reduce location accuracy to 20km, 10km etc(#181) by giving slightly wrong coordinates. This could be done by removing decimal places but that would cause many pictures to have the exact same coordinate. Another option would be adding a random number to the coordinate but if a user took enough pictures from his house, averaging all the coordinates would give their location. This would allow the user to choose between different sensitivities of geotagging instead of removing the feature(#1644). 

Additionally, a UI to edit EXIF data of a specific image from the ShareActivity could be implemented but I'm not sure how much of the upload workflow would have to be changed for that.

Also, what could be done about making location and timing information available for endangered species through the same feature? (#182) 

**Screen-shots:** 

![image](https://user-images.githubusercontent.com/7621539/42169067-656ebe4e-7e1b-11e8-97c6-1d64eaa05146.png)
![image](https://user-images.githubusercontent.com/7621539/42169094-7a5660dc-7e1b-11e8-8127-2524a520e204.png)
![image](https://user-images.githubusercontent.com/7621539/42171168-4433b4fe-7e21-11e8-8a93-2e4230350011.png)

**Would you like to work on the issue?**

Yes.

EDIT:I clarified what reducing GPS accuracy is.

**Comment by nicolas-raoul** [comment]
Endangered species pictures probably are less than 0.01% of all uploaded pictures, so if we implement some location-hiding it should definitely be per-picture rather than a general setting.

However, a hide-location-if-X-meters-from-home feature would have its place in the general settings :-)

What does it mean to "reduce accuracy" for a Commons picture?
Could you please include the URL to a Commons picture that has a location with reduced accuracy, so that we can have a look? Thanks! Or is it simply giving slightly wrong coordinates to the picture, for instance by removing decimals?

**Comment by ilgazer** [comment]
> Or is it simply giving slightly wrong coordinates to the picture, for instance by removing decimals?

Yeah, that's what I was thinking. I clarified the OP. Also, I agree that the endangered species thing isn't really a priority, but on #182 people proposed hiding the location if one of the categories you select is an endangered species. That would be seriously hard to implement though, as the upload starts before the category screen.

**Comment by ilgazer** [comment]
Oops, accidentally closed it. CAn someone reopen it?

**Comment by chrisdebian** [comment]
I can see the validity of this, to protect the contributor's privacy. Do we think it's still a useful enhancement?

Chris

**Comment by nicolas-raoul** [comment]
The app now has a setting to remove certain EXIF tags.
There is another issue about removing a picture's EXIF if desired.
Reducing coordinates precision around home sounds useful.
